packages/ssx-sdk/src/modules/Storage.ts

```diff
@@ -1,42 +1,217 @@
+   */
+  get(key: string): Promise<any>;
+
+  /**
+   * Stores a value with the specified key, encrypting it before storage.
+   * @param key The unique identifier for the stored value.
+   * @param value The value to store under the given key with encryption.
+   */
+  put(key: string, value: any): Promise<void>;
+}
+
+interface IStorageConfig {
+  prefix?: string;
+}
+
+interface IBrowserStorageConfig extends IStorageConfig {
+  storeName?: string;
+}
+
+class BrowserStorage implements IStorage {
+  private prefix: string;
+  private dbName: string;
+  private storeName: string;
+
+  constructor(config: IBrowserStorageConfig) {
+    // if no window object, throw error
+    if (typeof window === 'undefined') {
+      throw new Error('BrowserStorage: window object not found');
+    }
+
+    this.prefix = config?.prefix || '';
+    this.dbName = 'ssx-browser-storage';
+    this.storeName = config?.storeName || 'ssx-browser-storage-store';
+  }
+
+  private prefixKey(key: string): string {
+    return this.prefix ? `${this.prefix}/${key}` : key;
+  }
+
+  private async getDB(): Promise<IDBPDatabase> {
+    return await openDB(this.dbName, 1, {
+      upgrade(db) {
+        if (!db.objectStoreNames.contains('store')) {
+          db.createObjectStore('store');
+        }
+      },
+    });
+  }
+
+  public async get(key: string): Promise<any> {
+    const db = await this.getDB();
+    const tx = db.transaction('store', 'readonly');
+    const store = tx.objectStore('store');
+    const value = await store.get(this.prefixKey(key));
+    await tx.done;
+    return value;
+  }
+
+  public async put(key: string, value: any): Promise<void> {
+    const db = await this.getDB();
+    const tx = db.transaction('store', 'readwrite');
+    const store = tx.objectStore('store');
+    await store.put(value, this.prefixKey(key));
+    await tx.done;
+  }
+
+  public async list(): Promise<string[]> {
+    const db = await this.getDB();
+    const tx = db.transaction('store', 'readonly');
+    const store = tx.objectStore('store');
+    const keys = await store.getAllKeys();
+    await tx.done;
+    return keys as unknown as string[];
+  }
+
+  public async delete(key: string): Promise<void> {
+    const db = await this.getDB();
+    const tx = db.transaction('store', 'readwrite');
+    const store = tx.objectStore('store');
+    await store.delete(this.prefixKey(key));
+    await tx.done;
+  }
+
+  public async deleteAll(): Promise<void> {
+    const db = await this.getDB();
+    const tx = db.transaction('store', 'readwrite');
+    const store = tx.objectStore('store');
+    await store.clear();
+    await tx.done;
+  }
 }
 
-// class BrowserStorage implements IStorage {}
 // class BrowserDataVault extends BrowserStorage implements IDataVault {
-class BrowserDataVault implements IDataVault {
+class BrowserDataVault extends BrowserStorage implements IDataVault {
```

Comment:
Having the DataVault extend the Storage module is a really nice way to implement this